
# Hive报错：Error: FUNCTION 'NUCLEUS_ASCII' already exists. (state=X0Y68,code=30000) - Magician的博客 - CSDN博客


2018年08月30日 11:21:18[春雨里de太阳](https://me.csdn.net/qq_16633405)阅读数：218


Hive执行schematool -initSchema -dbType derby报错。
报错的日志：
```python
doupeihua@doupeihua-
```
```python
2104
```
```python
~/software/hive/bin $ schematool -initSchema -dbType derby
SLF4J:
```
```python
Class
```
```python
path contains multiple SLF4J bindings.
SLF4J: Found binding
```
```python
in
```
```python
[jar:
```
```python
file
```
```python
:/home/doupeihua/software/hive/lib/hive-jdbc-
```
```python
2.0
```
```python
.
```
```python
0
```
```python
-standalone.jar!/org/slf4j/impl/StaticLoggerBinder.
```
```python
class
```
```python
]
SLF4J: Found binding
```
```python
in
```
```python
[jar:
```
```python
file
```
```python
:/home/doupeihua/software/hive/lib/log4j-slf4j-impl-
```
```python
2.4
```
```python
.
```
```python
1
```
```python
.jar!/org/slf4j/impl/StaticLoggerBinder.
```
```python
class
```
```python
]
SLF4J: Found binding
```
```python
in
```
```python
[jar:
```
```python
file
```
```python
:/home/doupeihua/software/spark-
```
```python
1.6
```
```python
.
```
```python
2
```
```python
-bin-hadoop2.
```
```python
6
```
```python
/lib/spark-assembly-
```
```python
1.6
```
```python
.
```
```python
2
```
```python
-hadoop2.
```
```python
6.0
```
```python
.jar!/org/slf4j/impl/StaticLoggerBinder.
```
```python
class
```
```python
]
SLF4J: Found binding
```
```python
in
```
```python
[jar:
```
```python
file
```
```python
:/home/doupeihua/software/hadoop-
```
```python
2.6
```
```python
.
```
```python
4
```
```python
/share/hadoop/common/lib/slf4j-log4j12-
```
```python
1.7
```
```python
.
```
```python
5
```
```python
.jar!/org/slf4j/impl/StaticLoggerBinder.
```
```python
class
```
```python
]
SLF4J: See http:
```
```python
//www.slf4j.org/codes.html#multiple_bindings for an explanation.
```
```python
SLF4J: Actual binding is
```
```python
of
```
```python
type
```
```python
[org.apache.logging.slf4j.Log4jLoggerFactory]
Metastore connection URL:     jdbc:derby:;databaseName=metastore_db;create=true
Metastore Connection Driver :     org.apache.derby.jdbc.EmbeddedDriver
Metastore connection User:     APP
Starting metastore schema
```
```python
initialization
```
```python
to
```
```python
2.0
```
```python
.
```
```python
0
```
```python
Initialization
```
```python
script hive-schema-
```
```python
2.0
```
```python
.
```
```python
0
```
```python
.derby.sql
Error:
```
```python
FUNCTION
```
```python
'
```
```python
NUCLEUS_ASCII
```
```python
'
```
```python
already
```
```python
exists
```
```python
.
```
```python
(state=X0Y68,code=30000)
```
```python
org
```
```python
.
```
```python
apache
```
```python
.
```
```python
hadoop
```
```python
.
```
```python
hive
```
```python
.
```
```python
metastore
```
```python
.
```
```python
HiveMetaException
```
```python
:
```
```python
Schema
```
```python
initialization
```
```python
FAILED! Metastore state would be inconsistent !!
```
解决办法**（亲测有效）**：
**1、先将metastore_db更名为：metastore_db.tmp**
```python
/hive/bin
```
```python
$
```
```python
mv metastore_db metastore_db.tmp
```
**2、执行schematool -initSchema -dbType derby**
```python
hive/bin $ schematool
```
```python
-initSchema
```
```python
-dbType
```
```python
derby
```
显示过程如下则证明问题解决：
```python
SLF4J:
```
```python
Class path contains multiple SLF4J bindings.
```
```python
SLF4J:
```
```python
Found binding
```
```python
in
```
```python
[jar:file:/home/doupeihua/software/hive/lib/hive-jdbc-
```
```python
2.0
```
```python
.0
```
```python
-standalone
```
```python
.jar
```
```python
!/org/slf4j/impl/StaticLoggerBinder
```
```python
.class
```
```python
]
```
```python
SLF4J:
```
```python
Found binding
```
```python
in
```
```python
[jar:file:/home/doupeihua/software/hive/lib/log4j-slf4j-impl-
```
```python
2.4
```
```python
.1
```
```python
.jar
```
```python
!/org/slf4j/impl/StaticLoggerBinder
```
```python
.class
```
```python
]
```
```python
SLF4J:
```
```python
Found binding
```
```python
in
```
```python
[jar:file:/home/doupeihua/software/spark-
```
```python
1.6
```
```python
.2
```
```python
-bin-hadoop2
```
```python
.6
```
```python
/lib/spark-assembly-
```
```python
1.6
```
```python
.2
```
```python
-hadoop2
```
```python
.6
```
```python
.0
```
```python
.jar
```
```python
!/org/slf4j/impl/StaticLoggerBinder
```
```python
.class
```
```python
]
```
```python
SLF4J:
```
```python
Found binding
```
```python
in
```
```python
[jar:file:/home/doupeihua/software/hadoop-
```
```python
2.6
```
```python
.4
```
```python
/share/hadoop/common/lib/slf4j-log4j12-
```
```python
1.7
```
```python
.5
```
```python
.jar
```
```python
!/org/slf4j/impl/StaticLoggerBinder
```
```python
.class
```
```python
]
```
```python
SLF4J:
```
```python
See http://www
```
```python
.slf
```
```python
4j
```
```python
.org
```
```python
/codes
```
```python
.html
```
```python
#multiple_bindings for an explanation.
```
```python
SLF4J:
```
```python
Actual binding is of type [org
```
```python
.apache
```
```python
.logging
```
```python
.slf
```
```python
4j
```
```python
.Log
```
```python
4jLoggerFactory]
Metastore connection URL:     jdbc:derby:
```
```python
;databaseName=metastore_db;create=true
```
```python
Metastore Connection Driver :     org
```
```python
.apache
```
```python
.derby
```
```python
.jdbc
```
```python
.EmbeddedDriver
```
```python
Metastore connection User:     APP
Starting metastore schema initialization to
```
```python
2.0
```
```python
.0
```
```python
Initialization script hive-schema-
```
```python
2.0
```
```python
.0
```
```python
.derby
```
```python
.sql
```
```python
Initialization script completed
schemaTool completed
```
转自：[https://blog.csdn.net/doupeihua/article/details/52496499](https://blog.csdn.net/doupeihua/article/details/52496499)

